torch_tvm/compiler.cpp

```diff
@@ -10,24 +10,36 @@ using namespace torch::jit;
 
 tvm::relay::Var TVMCompiler::convertToRelay(Value* val, TVMContext ctx) {
   auto optional_ivalue = toIValue(val);
-  if (optional_ivalue.has_value()) {
+
+  tvm::Array<HalideIR::Expr> sizes;
+  if (auto ptt = val->type()->cast<ProfiledTensorType>())
+  {
+
+    auto csizes = ptt->sizes().concrete_sizes();
+    TORCH_INTERNAL_ASSERT(csizes.has_value());
```

Comment:
should we handle the case when this isn't true?

Reply:
I think we shouldn't just create a TVMCompGroup in this case. Even though, we can compile at a runtime if we get a workload where shapes change all the time it will be pretty wasteful. I'm looking into changing Fuser to not fuse things that don't have `ProfiledTensorTypes`

Reply:
if it's alternating between 20 batch size shapes but run 1M times it's probably worth still compiling each shape

Reply:
hopefully, this will be handled via bailouts. We specialize for shape_set1. Then if we see another frequent set, we will specialize for that one as well and so on.